{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Retriving all RNA-seq experiments from SRA\n",
    "\n",
    "I used pysradb to retrieve all RNA-seq experiments from SRA. I used the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! pysradb search \"rna-seq Populus tremula X Populus alba\" --saveto Populus_tremula_X_Populus_alba.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# some data wrangling\n",
    "\n",
    "some studies were poorly annotated and I had to know how many samples were in the table. I used the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "sra_table = pd.read_csv(\"Populus_tremula_X_Populus_alba.tsv\", sep=\"\\t\", index_col=0)\n",
    "print(sra_table.shape)\n",
    "tree_sra_table = sra_table[sra_table.organism_name == \"Populus tremula x Populus alba\"]\n",
    "print(tree_sra_table.shape)\n",
    "print(len(tree_sra_table.index.unique()))\n",
    "tree_sra_table.value_counts(\"experiment_desc\")\n",
    "print(tree_sra_table.value_counts(\"study_accession\")) # used this to find out some study accessions have only one sample"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# script generation - download all samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"sbatch.sh\", 'w') as ff:\n",
    "    for SRP in tree_sra_table.index.unique():\n",
    "        ff.write(\"sbatch \" + f\"{SRP}.sh\" + \"\\n\")\n",
    "        with open(f\"{SRP}.sh\", 'w') as f:\n",
    "            f.write(f\"\"\"#!/bin/bash\n",
    "#SBATCH --job-name=pysradb         # Job name\n",
    "#SBATCH --partition=highmem_p             # Partition (queue) name\n",
    "#SBATCH --ntasks=1            # Run on a single CPU\n",
    "#SBATCH --cpus-per-task=8\n",
    "#SBATCH --mem=100gb                     # Job memory request\n",
    "#SBATCH --time=168:00:00               # Time limit hrs:min:sec\n",
    "#SBATCH --output=sra.%j.out    # Standard output log\n",
    "#SBATCH --error=sra.%j.err     # Standard error log\n",
    "#SBATCH --mail-type=END,FAIL          # Mail events (NONE, BEGIN, END, FAIL, ALL)\n",
    "#SBATCH --mail-user=ch29576@uga.edu  # Where to send mail\n",
    "\n",
    "date\n",
    "cd $SLURM_SUBMIT_DIR\n",
    "pysradb download -y -t 8 -p {SRP}\"\"\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# script generation - from .sra to .fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"sbatch_sra2fastq.sh\", 'w') as ff:\n",
    "    for SRP in tree_sra_table.index.unique():\n",
    "        ff.write(\"sbatch \" + f\"./script/{SRP}_sra2fastq.sh\" + \"\\n\")\n",
    "        with open(f\"./script/{SRP}_sra2fastq.sh\", 'w') as f:\n",
    "            f.write(f\"\"\"#!/bin/bash\n",
    "#SBATCH --job-name=pysradb         # Job name\n",
    "#SBATCH --partition=highmem_p             # Partition (queue) name\n",
    "#SBATCH --ntasks=1            # Run on a single CPU\n",
    "#SBATCH --cpus-per-task=8\n",
    "#SBATCH --mem=100gb                     # Job memory request\n",
    "#SBATCH --time=168:00:00               # Time limit hrs:min:sec\n",
    "#SBATCH --output=sra.%j.out    # Standard output log\n",
    "#SBATCH --error=sra.%j.err     # Standard error log\n",
    "#SBATCH --mail-type=ALL          # Mail events (NONE, BEGIN, END, FAIL, ALL)\n",
    "#SBATCH --mail-user=ch29576@uga.edu  # Where to send mail\n",
    "\n",
    "date\n",
    "ml SRA-Toolkit\n",
    "cd $SLURM_SUBMIT_DIR\n",
    "\"\"\")\n",
    "            current_df = tree_sra_table.loc[SRP]\n",
    "            if type(current_df) != pd.core.series.Series:\n",
    "                for SRP_,info in current_df.iterrows():\n",
    "                    f.write(f\"fasterq-dump --split-3 ./{SRP_}/{info.experiment_accession}/{info.run_accession}.sra -O ./{SRP_}/fastq\\n\")\n",
    "                \n",
    "            else:\n",
    "                f.write(f\"fasterq-dump --split-3 ./{SRP}/{current_df.experiment_accession}/{current_df.run_accession}.sra -O ./{SRP}/fastq\\n\")\n",
    "            f.write(f\"date\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# script generation - prepare files for nfcore/rnaseq pipeline\n",
    "\n",
    "here requires some manual work. some samples are not paired-end and some samples were not converted correctly in the previous step. I have to edit the file manually for a few cases."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"arrange_nfcore_rnaseq.sh\", 'w') as ff:\n",
    "    for SRP in tree_sra_table.index.unique():\n",
    "\n",
    "        # # gzip all fastq\n",
    "        # ff.write(f\"cd {SRP}/fastq\\n\")\n",
    "        # ff.write(f\"rm -rf run.sh\\n\")\n",
    "        # ff.write(f\"\"\"echo '#!/bin/sh' > run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH -J {SRP}_gzip' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --partition batch' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --ntasks=1' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --cpus-per-task=8' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --time=124:00:00' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --mem=16gb' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --mail-user=ch29576@uga.edu' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '#SBATCH --mail-type=ALL' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo 'date' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo 'cd $SLURM_SUBMIT_DIR' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo 'gzip *.fastq' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo 'date' >> run.sh\\n\"\"\")\n",
    "        # ff.write(f\"sbatch run.sh\\n\")\n",
    "        # ff.write(f\"cd ../../\\n\")\n",
    "\n",
    "        # create input.csv\n",
    "        ff.write(f\"cd {SRP}\\n\")\n",
    "        ff.write(f\"rm -rf input.csv\\n\")\n",
    "        ff.write(f\"echo 'sample,fastq_1,fastq_2,strandedness' > input.csv\\n\")\n",
    "        current_df = tree_sra_table.loc[SRP]\n",
    "        if type(current_df) != pd.core.series.Series:\n",
    "            if current_df[\"library_layout\"][0] == \"SINGLE\":\n",
    "                for SRP_,info in current_df.iterrows():\n",
    "                    ff.write(f\"\"\"echo '{info.run_accession},./fastq/{info.run_accession}.fastq.gz,,unstranded' >> input.csv\\n\"\"\")\n",
    "    \n",
    "        #     else:\n",
    "        #         for SRP_,info in current_df.iterrows():\n",
    "        #             ff.write(f\"\"\"echo '{info.run_accession},./fastq/{info.run_accession}_1.fastq.gz,./fastq/{info.run_accession}_2.fastq.gz,unstranded' >> input.csv\\n\"\"\")\n",
    "        # else:\n",
    "        #     ff.write(f\"\"\"echo '{current_df.run_accession},./fastq/{current_df.run_accession}_1.fastq.gz,./fastq/{current_df.run_accession}_2.fastq.gz,unstranded' >> input.csv\\n\"\"\")\n",
    "        ff.write(f\"cd ..\\n\")\n",
    "\n",
    "        # # create nf-params.json\n",
    "        # ff.write(f\"cd {SRP}\\n\")\n",
    "        # ff.write(f\"rm -rf nf-params.json\\n\")\n",
    "        # ff.write(f\"mkdir results\\n\")\n",
    "        # # let the shell script creaete a file with the following content\n",
    "        # # {\n",
    "        # #     \"input\": \".\\/input.csv\",\n",
    "        # #     \"outdir\": \".\\/results\",\n",
    "        # #     \"email\": \"ch29576@uga.edu\",\n",
    "        # #     \"multiqc_title\": \"2022SUT4\",\n",
    "        # #     \"fasta\": \"\\/scratch\\/ch29576\\/v5_717\\/reference\\/717v5.fa\",\n",
    "        # #     \"gff\": \"\\/scratch\\/ch29576\\/v5_717\\/reference\\/717v5.gff\",\n",
    "        # #     \"gtf_extra_attributes\": \"Name\",\n",
    "        # #     \"featurecounts_group_type\": \"\",\n",
    "        # #     \"featurecounts_feature_type\": \"\",\n",
    "        # #     \"skip_biotype_qc\": true,\n",
    "        # #     \"pseudo_aligner\": \"salmon\",\n",
    "        # #     \"aligner\": \"star_salmon\"\n",
    "        # # }\n",
    "        # ff.write(f\"echo '{{' > nf-params.json\\n\")\n",
    "        # ff.write(f\"\"\"echo '    \"input\": \".\\/input.csv\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"outdir\": \".\\/results\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"email\": \"ch29576@uga.edu\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"multiqc_title\": \"{SRP}\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"fasta\": \"\\/scratch\\/ch29576\\/v5_717\\/reference\\/717v5.fa\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"gff\": \"\\/scratch\\/ch29576\\/v5_717\\/reference\\/717v5.gff\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"gtf_extra_attributes\": \"Name\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"featurecounts_group_type\": \"\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"featurecounts_feature_type\": \"\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"skip_biotype_qc\": true,' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"pseudo_aligner\": \"salmon\",' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"\"\"echo '    \"aligner\": \"star_salmon\"' >> nf-params.json\\n\"\"\")\n",
    "        # ff.write(f\"echo '}}' >> nf-params.json\\n\")\n",
    "        # ff.write(f\"cd ..\\n\")\n",
    "\n",
    "        # create run.sh\n",
    "        ff.write(f\"cd {SRP}\\n\")\n",
    "        ff.write(f\"rm -rf run.sh\\n\")\n",
    "        ff.write(f\"\"\"echo '#!/bin/sh' > run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH -J {SRP}_RNA-Seq' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --partition highmem_p' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --ntasks=1' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --cpus-per-task=28' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --time=168:00:00' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --mem=400gb' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --mail-user=ch29576@uga.edu' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo '#SBATCH --mail-type=ALL' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'date' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'cd $SLURM_SUBMIT_DIR' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'ml Nextflow' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'NXF_SINGULARITY_CACHEDIR=/scratch/ch29576/singularity_cache' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'nextflow run nf-core/rnaseq -r 3.8.1 -name {SRP}-SE --outdir results -profile singularity -params-file nf-params.json -resume ' >> run.sh\\n\"\"\")\n",
    "        ff.write(f\"\"\"echo 'date' >> run.sh\\n\"\"\")\n",
    "        if current_df[\"library_layout\"][0] == \"SINGLE\":\n",
    "            ff.write(f\"sbatch run.sh\\n\")\n",
    "        ff.write(f\"cd ..\\n\")\n",
    "        "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metadata from bioPython\n",
    "\n",
    "I tried to use pysradb to get metadata but did not get what I want. So here I use bioPython to get the study abstract, which provided a little bit more info fo reach study."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "! pysradb metadata SRP364200 --detailed --saveto SRP364200_metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EXTERNAL_ID</th>\n",
       "      <th>STUDY_TITLE</th>\n",
       "      <th>STUDY_ABSTRACT</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRP364200</th>\n",
       "      <td>PRJNA631840</td>\n",
       "      <td>Populus tremula x Populus alba Transcriptome o...</td>\n",
       "      <td>Transcriptome in in Poplar in response to nitr...</td>\n",
       "      <td>Chinese Academy of Forestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP350268</th>\n",
       "      <td>GSM5727288</td>\n",
       "      <td>Single nuclei transcriptomic of Populus tremul...</td>\n",
       "      <td>Here we applied a novel approach to isolate nu...</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP349172</th>\n",
       "      <td>PRJNA785328</td>\n",
       "      <td>717-1B4 XBAT35 Root RNA-Seq</td>\n",
       "      <td>Transgenic 717-1B4 XBAT35 RNA-Seq from 12d root</td>\n",
       "      <td>Oak Ridge National Laboratory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP343659</th>\n",
       "      <td>GSM5660932</td>\n",
       "      <td>Transcriptome profiling of glycosyltransferase...</td>\n",
       "      <td>Comparison of UGT71L1 knock-out genes with emp...</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP331923</th>\n",
       "      <td>PRJNA753499</td>\n",
       "      <td>RNAseq data from poplar leaf development</td>\n",
       "      <td>RNAseq data of poplar leaf from three developm...</td>\n",
       "      <td>University of Georgia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP059838</th>\n",
       "      <td>PRJNA275366</td>\n",
       "      <td>Populus: Defoliation experiment</td>\n",
       "      <td>Carbon partitioning in poplar plant.</td>\n",
       "      <td>University of Georgia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP042117</th>\n",
       "      <td>PRJNA247385</td>\n",
       "      <td>Xylem transcriptome response of transgenic Pop...</td>\n",
       "      <td>Wild type and transgenic Populus plants were s...</td>\n",
       "      <td>University of Georgia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP058772</th>\n",
       "      <td>PRJNA284622</td>\n",
       "      <td>Populus tremula x Populus alba cultivar:Populu...</td>\n",
       "      <td>Study of gene expression in different wood typ...</td>\n",
       "      <td>USDA FOREST SERVICE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP058044</th>\n",
       "      <td>GSM1676548</td>\n",
       "      <td>Changes in gene expression in the bark of MIR15</td>\n",
       "      <td>Illumina HiSeq technology was used to generate...</td>\n",
       "      <td>Gene Expression Omnibus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRP041243</th>\n",
       "      <td>GSM1370395</td>\n",
       "      <td>Effect of Methyl jasmonate of the Poplar root ...</td>\n",
       "      <td>Illumina HiSeq technology was used to generate...</td>\n",
       "      <td>Gene Expression Omnibus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>83 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           EXTERNAL_ID                                        STUDY_TITLE  \\\n",
       "SRP                                                                         \n",
       "SRP364200  PRJNA631840  Populus tremula x Populus alba Transcriptome o...   \n",
       "SRP350268   GSM5727288  Single nuclei transcriptomic of Populus tremul...   \n",
       "SRP349172  PRJNA785328                        717-1B4 XBAT35 Root RNA-Seq   \n",
       "SRP343659   GSM5660932  Transcriptome profiling of glycosyltransferase...   \n",
       "SRP331923  PRJNA753499           RNAseq data from poplar leaf development   \n",
       "...                ...                                                ...   \n",
       "SRP059838  PRJNA275366                    Populus: Defoliation experiment   \n",
       "SRP042117  PRJNA247385  Xylem transcriptome response of transgenic Pop...   \n",
       "SRP058772  PRJNA284622  Populus tremula x Populus alba cultivar:Populu...   \n",
       "SRP058044   GSM1676548    Changes in gene expression in the bark of MIR15   \n",
       "SRP041243   GSM1370395  Effect of Methyl jasmonate of the Poplar root ...   \n",
       "\n",
       "                                              STUDY_ABSTRACT  \\\n",
       "SRP                                                            \n",
       "SRP364200  Transcriptome in in Poplar in response to nitr...   \n",
       "SRP350268  Here we applied a novel approach to isolate nu...   \n",
       "SRP349172    Transgenic 717-1B4 XBAT35 RNA-Seq from 12d root   \n",
       "SRP343659  Comparison of UGT71L1 knock-out genes with emp...   \n",
       "SRP331923  RNAseq data of poplar leaf from three developm...   \n",
       "...                                                      ...   \n",
       "SRP059838               Carbon partitioning in poplar plant.   \n",
       "SRP042117  Wild type and transgenic Populus plants were s...   \n",
       "SRP058772  Study of gene expression in different wood typ...   \n",
       "SRP058044  Illumina HiSeq technology was used to generate...   \n",
       "SRP041243  Illumina HiSeq technology was used to generate...   \n",
       "\n",
       "                                    Name  \n",
       "SRP                                       \n",
       "SRP364200    Chinese Academy of Forestry  \n",
       "SRP350268                           NCBI  \n",
       "SRP349172  Oak Ridge National Laboratory  \n",
       "SRP343659                           NCBI  \n",
       "SRP331923          University of Georgia  \n",
       "...                                  ...  \n",
       "SRP059838          University of Georgia  \n",
       "SRP042117          University of Georgia  \n",
       "SRP058772            USDA FOREST SERVICE  \n",
       "SRP058044        Gene Expression Omnibus  \n",
       "SRP041243        Gene Expression Omnibus  \n",
       "\n",
       "[83 rows x 4 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read all SRP and query the study abstract & save to csv\n",
    "import Bio\n",
    "from Bio import Entrez\n",
    "import bs4 as bs\n",
    "import requests\n",
    "import lxml\n",
    "import pandas as pd\n",
    "from tqdm import tqdm\n",
    "\n",
    "sra_table = pd.read_csv(\"Populus_tremula_X_Populus_alba.tsv\", sep=\"\\t\", index_col=0)\n",
    "tree_sra_table = sra_table[sra_table.organism_name == \"Populus tremula x Populus alba\"]\n",
    "tree_sra_table\n",
    "SRP_list = list(tree_sra_table.index.unique())\n",
    "\n",
    "attr_list = [\n",
    "    'EXTERNAL_ID',\n",
    "    'STUDY_TITLE',\n",
    "    'STUDY_ABSTRACT',\n",
    "    # 'SUBMITTER_ID',\n",
    "    'Name',\n",
    "]\n",
    "srp_df = pd.DataFrame(columns=(attr_list + [\"SRP\"]))\n",
    "srp_df.set_index(\"SRP\", inplace=True)\n",
    "Entrez.email = \"jgi@gmail.com\"\n",
    "\n",
    "# query the study abstract\n",
    "for SRP in tqdm(SRP_list):\n",
    "\n",
    "    \n",
    "    handle = Entrez.esearch(db=\"sra\", term=SRP, retmax=100000)\n",
    "    record = Entrez.read(handle)\n",
    "    handle.close()\n",
    "    for id in record[\"IdList\"]:\n",
    "        handle = Entrez.efetch(db=\"sra\", id=id)\n",
    "        record = handle.read()\n",
    "        handle.close()\n",
    "    file = bs.BeautifulSoup(record, features=\"xml\")\n",
    "    for attr in attr_list:\n",
    "        try:\n",
    "            srp_df.loc[SRP, attr] = file.find(attr).get_text()\n",
    "        except:\n",
    "            srp_df.loc[SRP, attr] = \"NA\"\n",
    "    \n",
    "\n",
    "srp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "srp_df.to_csv(\"Populus_tremula_X_Populus_alba_srp_abstract.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "9abd3d8c29d05eab696a56da028975b41e3b8553b095cdd99200e9c14a92c067"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
